### PR TITLE
2.0.0 Test-ProGetFeed fails if session missing api key

### PR DESCRIPTION
**Pull request by splatteredbits**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,82 @@
...
-## Added
+### Upgrade Instructions
+
+Updated `Get-ProGetFeed`, `New-ProGetFeed`, `Test-ProGetFeed`, and `Remove-ProGetFeed` functions to use ProGet's
+[Feed Management API](https://docs.inedo.com/docs/proget-reference-api-feed-management) instead of the native API.
+Ensure that API keys/credentials used with those functions have appropriate access to manage feeds.
+
+Update usages of objects returned by `Get-ProGetFeed` and `New-ProGetFeed`. Rename these properties:
+
+* `Active_Indicator` -> `active`
```
Comment (decoyjoe) [inline CHANGELOG.md:15]: I like to use ➔ ([Unicode Character U+2794](https://www.compart.com/en/unicode/U+2794))

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,9 +1,82 @@
...
+* `AllowUnknownLicenses_Indicator` -> `allowUnknownLicenses`
+* `Feed_Description` -> `description`
+* `AllowUnassessedVulnerabilities_Indicator` ->
```
Comment (decoyjoe) [inline CHANGELOG.md:18]: Missing new property name on the right side of the arrow.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,9 +1,82 @@
...
+if a feed exists. All feed names, regardless of type, must now be unique.
+
+Make sure `Delete` HTTP verbs are allowed to your instance of ProGet. The `Remove-ProGetFeed` function now uses that
+verb when deleting a feed.
+
+Add `-ErrorAction Ignore` to usages of `Get-ProGetFeed`. That function now writes an error if a feed does not exist.
+
+### Added
+
+* The `Publish-ProGetUniversalPackage` can now authenticate using a ProGet API key.
```
Comment (decoyjoe) [inline CHANGELOG.md:49]: Missing the word "function"

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,9 +1,82 @@
+<!--markdownlint-disable MD024 no-duplicate-header-->
 
 # ProGetAutomation Changelog
 
-# 1.1.0
+## 2.0.0
```
Comment (decoyjoe) [inline CHANGELOG.md:5]: Nit: Add release date.

File: ProGetAutomation/Functions/Get-ProGetFeed.ps1 [hunk 5]
```diff
@@ -6,11 +6,15 @@ function Get-ProGetFeed
...
+     the `Session` parameter. Use `New-ProGetSession` to create a session. By default, only active feeds are returned.
+     Use the `-Force` switch to also return inactive feeds.
 
-    To get a specific feed, pass its name to the `Name` parameter. If the feed by that name doesn't exist, nothing is returned and no errors are written. 
+    To get a specific feed, pass its name to the `Name` parameter. If the feed by that name doesn't exist, nothing is
+    returned and no errors are written.
 
-    This function uses the `Feeds_GetFeed` and `Feeds_GetFeeds` endpoints in ProGet's [native API](https://inedo.com/support/documentation/proget/reference/api/native).
+    This function uses the `Feeds_GetFeed` and `Feeds_GetFeeds` endpoints in ProGet's
+    [native API](https://inedo.com/support/documentation/proget/reference/api/native).
```
Comment (decoyjoe) [inline ProGetAutomation/Functions/Get-ProGetFeed.ps1:17]: Docs needs updating, the function now uses the feed management API.

File: ProGetAutomation/Functions/Invoke-ProGetRestMethod.ps1 [hunk 6]
```diff
@@ -6,58 +6,59 @@ function Invoke-ProGetRestMethod
...
+        [switch] $Raw
     )
 
     Set-StrictMode -Version 'Latest'
     Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
 
     $uri = New-Object 'Uri' -ArgumentList $Session.Url,$Path
 
-    $requestContentType = 'application/json; charset=utf-8'
+    $requestContentType = 'application/json' #; charset=utf-8'
```
Comment (decoyjoe) [inline ProGetAutomation/Functions/Invoke-ProGetRestMethod.ps1:61]: Intentional comment or debugging artifact?

File: ProGetAutomation/Functions/New-ProGetFeed.ps1 [hunk 7]
```diff
@@ -6,12 +6,32 @@ function New-ProGetFeed
     Creates a new ProGet package feed
 
     .DESCRIPTION
-    The `New-ProGetFeed` function creates a new ProGet feed. Use the `Type` parameter to specify the feed type (valid values are 'VSIX', 'RubyGems', 'Docker', 'ProGet', 'Maven', 'Bower', 'npm', 'Deployment', 'Chocolatey', 'NuGet', 'PowerShell'). The `Session` parameter controls the instance of ProGet to connect to. This function uses ProGet's Native API, so an API key is required. Use `New-ProGetSession` to create a session with your API key.
+    The `New-ProGetFeed` function creates a new ProGet feed. Use the `Type` parameter to specify the feed type. The
+    `Session` parameter controls the instance of ProGet to connect to. This function uses ProGet's Native API, so an API
+    key is required. Use `New-ProGetSession` to create a session with your API key.
```
Comment (decoyjoe) [inline ProGetAutomation/Functions/New-ProGetFeed.ps1:11]: Docs need updating, function no longer uses the native API.

File: ProGetAutomation/Functions/Remove-ProGetFeed.ps1 [hunk 8]
```diff
@@ -5,12 +5,18 @@ function Remove-ProGetFeed
     .SYNOPSIS
     Removes a feed from ProGet.
 
-    .DESCRIPTION
-    The `Remove-ProGetFeed` function removes a feed from ProGet. All packages in the feed are also deleted. Pass the session to the ProGet instance from which to delete the feed to the `Session` parameter (use the `New-ProGetSession` function to create a session. Pass the ID of the feed to the `ID` parameter. You can also pipe feed IDs or feed objects returned by `Get-ProGetFeed`.
+    .continueAndInquireMsg
```
Comment (decoyjoe) [inline ProGetAutomation/Functions/Remove-ProGetFeed.ps1:8]: What's `.continueAndInquireMsg` ?

File: ProGetAutomation/Functions/Remove-ProGetFeed.ps1 [hunk 9]
```diff
@@ -5,12 +5,18 @@ function Remove-ProGetFeed
...
+     objects returned by `Get-ProGetFeed`.
 
-    Since this has the potential to be a disastrous operation (did we mention all the packages in the feed will also get deleted) and can't be undone, you'll be asked to confirm the deletion. If you don't want to be prompted, use the `-Force` switch. This is dangerous.
+    Since this has the potential to be a disastrous operation (did we mention all the packages in the feed will also get
+    deleted) and can't be undone, you'll be asked to confirm the deletion. If you don't want to be prompted, use the
+    `-Force` switch. This is dangerous.
 
-    This function uses the `Feeds_DeleteFeed` endpoint in [ProGet's native API](https://inedo.com/support/documentation/proget/reference/api/native).
+    This function uses the `Feeds_DeleteFeed` endpoint in
+    [ProGet's native API](https://inedo.com/support/documentation/proget/reference/api/native).
```
Comment (decoyjoe) [inline ProGetAutomation/Functions/Remove-ProGetFeed.ps1:19]: Function now uses the feed management API.

**Review by decoyjoe — APPROVED** [review]

File: ProGetAutomation/Functions/Invoke-ProGetRestMethod.ps1 (hunk 6 above)
Comment (splatteredbits) [reply ProGetAutomation/Functions/Invoke-ProGetRestMethod.ps1:61]: API was throwing exceptions if the charset directive was present. I reported it as a bug ,with the intent to uncomment when it was fixed, but everything works with it missing, so I'll remove.

File: ProGetAutomation/Functions/Remove-ProGetFeed.ps1 (hunk 8 above)
Comment (splatteredbits) [reply ProGetAutomation/Functions/Remove-ProGetFeed.ps1:8]: Visual Studio Code's "Rename" functionality (Ctrl+F2), which essentially just does a search/replace. :(

File: CHANGELOG.md (hunk 1 above)
Comment (splatteredbits) [reply CHANGELOG.md:15]: Do you have an extension or keybinding for that? Usually I use Win+., but think it's annoying.

Comment (decoyjoe) [reply CHANGELOG.md]: Yes I use AutoHotKey [Hotstrings](https://www.autohotkey.com/docs/v1/Hotstrings.htm) to do text expansion. For this character I have mapped `{backtick}{backtick}rightb{space}` which results in AutoHotKey replacing that sequence with the ➔ character.

Here's my current list of hotstrings for reference:

```ahk
; ```` == two typed backticks (a backtick is an AHK escape character)
::````check::✓
::````enter::⏎
::````right::→
::````rightb::➔
::````rb::➔
::````left::←
::````up::↑
::````down::↓
::````tm::™
::````em::—
```

**Pull request merged**